File: plugins/hamsurang-slide/skills/hamsurang-slide/scripts/inline_assets.mjs
#!/usr/bin/env node
import { readFileSync, writeFileSync, existsSync } from 'node:fs'
import { basename, resolve } from 'node:path'

const svgToDataUri = (svgPath) => {
  const data = readFileSync(svgPath)
  return `data:image/svg+xml;base64,${data.toString('base64')}`
}

const svgToInline = (svgPath) => readFileSync(svgPath, 'utf8').trim()

const main = () => {
  if (process.argv.length < 4) {
    console.log(`Usage: ${basename(process.argv[1])} <input.html> <images_dir>`)
    process.exit(1)
  }

  const htmlPath = resolve(process.argv[2])
  const imagesDir = resolve(process.argv[3])

  let html = readFileSync(htmlPath, 'utf8')

  const replacements = new Map([
    ['{{LOGO_DATA_URI}}', ['logo.svg', svgToDataUri]],
    ['{{LOGO_WHITE_DATA_URI}}', ['logo-white.svg', svgToDataUri]],
    ['{{MOUNTAIN_SVG_INLINE}}', ['mountain.svg', svgToInline]],
    ['{{MOUNTAIN_WHITE_SVG_INLINE}}', ['mountain-white.svg', svgToInline]],
  ])

  for (const [placeholder, [filename, converter]] of replacements) {
    const filePath = resolve(imagesDir, filename)
    if (existsSync(filePath) && html.includes(placeholder)) {
      html = html.replaceAll(placeholder, converter(filePath))
      console.log(`  Replaced ${placeholder} from ${filename}`)
      continue
    }

    if (html.includes(placeholder)) {
      console.log(`  WARNING: ${filename} not found, ${placeholder} left as-is`)
    }
  }

  writeFileSync(htmlPath, html, 'utf8')
  console.log(`Done: ${htmlPath}`)
}

main()

File: plugins/hamsurang-slide/skills/hamsurang-slide/scripts/inline_assets.py
#!/usr/bin/env python3
import base64
import sys
from pathlib import Path


def svg_to_data_uri(svg_path: Path) -> str:
    data = svg_path.read_bytes()
    b64 = base64.b64encode(data).decode()
    return f"data:image/svg+xml;base64,{b64}"


def svg_to_inline(svg_path: Path) -> str:
    return svg_path.read_text(encoding="utf-8").strip()


def main():
    if len(sys.argv) < 3:
        print(f"Usage: {sys.argv[0]} <input.html> <images_dir>")
        sys.exit(1)

    html_path = Path(sys.argv[1])
    images_dir = Path(sys.argv[2])

    html = html_path.read_text(encoding="utf-8")

    replacements = {
        "{{LOGO_DATA_URI}}": ("logo.svg", svg_to_data_uri),
        "{{LOGO_WHITE_DATA_URI}}": ("logo-white.svg", svg_to_data_uri),
        "{{MOUNTAIN_SVG_INLINE}}": ("mountain.svg", svg_to_inline),
        "{{MOUNTAIN_WHITE_SVG_INLINE}}": ("mountain-white.svg", svg_to_inline),
    }

    for placeholder, (filename, converter) in replacements.items():
        filepath = images_dir / filename
        if filepath.exists() and placeholder in html:
            html = html.replace(placeholder, converter(filepath))
            print(f"  Replaced {placeholder} from {filename}")
        elif placeholder in html:
            print(f"  WARNING: {filename} not found, {placeholder} left as-is")

    html_path.write_text(html, encoding="utf-8")
    print(f"Done: {html_path}")


if __name__ == "__main__":
    main()
